### PR TITLE
Feature/broadcast new blocks 507

### PR DESCRIPTION
**Pull request by zilm13**
Broadcast new txs
Broadcast new blocks/hashes
Resend txs on disconnect
Test

### CODE REVIEW

**Comment by zilm13** [comment]
Removed #526 #527 as duplicates

File: ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java [hunk 1]
```diff
@@ -204,7 +235,7 @@ private void blockRetrieveLoop() {
                         // to get more chances to receive block body promptly
                         for (BlockHeaderWrapper blockHeaderWrapper : bReq.getBlockHeaders()) {
                             Channel channel = pool.getByNodeId(blockHeaderWrapper.getNodeId());
-                            if (channel != null) {
+                            if (channel != null && channel.isIdle()) {
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java]: Sometimes cause peer drop with reputation reduce.
Actually we have no guarantee in multi-thread env on the next few lines but had not ever run it in.

File: ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java [hunk 2]
```diff
@@ -269,10 +316,61 @@ private void produceQueue() {
...
+                    logger.error("Error processing unknown block", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Processing new blocks from queue
+     */
+    private void newBlocksDistributeLoop() {
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java]: M.b. let's move blocks and tx broadcasting logic from `SyncManager` to let's say `ChannelManager`. 
Those functions has actually few in common with syncing and more with networking

File: ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java [hunk 3]
```diff
@@ -91,9 +104,8 @@ public void accept(BlockWrapper blockWrapper) {
     private CountDownLatch receivedHeadersLatch = new CountDownLatch(0);
     private CountDownLatch receivedBlocksLatch = new CountDownLatch(0);
 
-    private Thread syncQueueThread;
-    private Thread getHeadersThread;
-    private Thread getBodiesThread;
+    // All workers-like thread loops to simplify shutdown
+    private List<Thread> threadLoops = new ArrayList<>();
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java]: I would also return back these fields to simplify further class reading. (each of these threads has its own essential destiny :) )

**Comment by zilm13** [comment]
@Nashatyrev please, take a look!

**Comment by Nashatyrev** [comment]
Cool, thanks!

**Pull request merged**